add support for java data loaders (#846)
observablehq/framework: docs/loaders.md

- `.R` - R (`Rscript`)
- `.rs` - Rust (`rust-script`)
- `.go` - Go (`go run`)

> - `.java` - Java >=11 (`java`)

Reviewer: ```suggestion
- `.java` - Java (`java`; requires Java 11+ and [single-file programs](https://openjdk.org/jeps/330))
```

- `quakes.csv.R`
- `quakes.csv.rs`
- `quakes.csv.go`
- `quakes.csv.java`
- `quakes.csv.sh`
- `quakes.csv.exe`

> If you use `.py`, `.R`, `.rs`, `.go` or `java`, the corresponding interpreter (`python3`, `Rscript`, `rust-script`, `go run` or `java`, respectively) must be installed and available on your `$PATH`. Any additional modules, packages, libraries, _etc._, must also be installed before you can use them.

Reviewer: ```suggestion
If you use `.py`, `.R`, `.rs`, `.go`, or `java`, the corresponding interpreter (`python3`, `Rscript`, `rust-script`, `go run`, or `java`, respectively) must be installed and available on your `$PATH`. Any additional modules, packages, libraries, _etc._, must also be installed before you can use them.
```

> Whereas `.js`, `.ts`, `.py`, `.R`, `.rs`, `.go`, `.java` and `.sh` data loaders are run via interpreters, `.exe` data loaders are run directly and must have the executable bit set. This is typically done via [`chmod`](https://en.wikipedia.org/wiki/Chmod). For example:

Reviewer: ```suggestion
Whereas all other data loaders are run via interpreters (such as `node` for `.js`), `.exe` data loaders are run directly and must have the executable bit set. This is typically done via [`chmod`](https://en.wikipedia.org/wiki/Chmod). For example:
```

I think we have enough data loaders now that it makes sense to refer to them in aggregate and only call out `.exe` is being special.
